### PR TITLE
prepare next ride feature

### PR DESCRIPTION
**Pull request by mltbnz**
In preparation of the next ride feature I added a `Ride` struct and the request to get the next rides.
Had to update the `APIRequestDefining` protocol to support queryItems.

I think to have a nice UX/UI for the user we should build some sort of toggle to display the nextRide. Also maybe the user wants to configure how in which radius the app should request rides and things like that.

First steps to #194 

### Before merging the PR

- [ ] If applicable, does the PR contain enough unit / UI tests?
- [ ] If applicable, did you add a before / after screenshot of the feature?
- [ ] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: CriticalMass/APIRequest.swift [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
 public protocol APIRequestDefining {
     associatedtype ResponseDataType: Decodable
     var endpoint: Endpoint { get }
     var httpMethod: HTTPMethod { get }
     var headers: HTTPHeaders? { get }
     var requiresBackgroundTask: Bool { get }
-    func makeRequest() -> URLRequest
+    func makeRequest() throws -> URLRequest
     func parseResponse(data: Data) throws -> ResponseDataType
+    func getQueryItems() -> [URLQueryItem]?
```
Comment (lennet) [inline CriticalMass/APIRequest.swift]: Why is GetQueryItems a function and not a variable?

File: CriticalMass/MapViewController.swift [hunk 2]
```diff
@@ -178,6 +184,20 @@ class MapViewController: UIViewController {
     @objc func didReceiveInitialLocation(notification: Notification) {
         guard let location = notification.object as? Location else { return }
         focusOnLocation(location: location)
+
+        guard Feature.events.isActive else { return }
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/MapViewController.swift]: Think so too. I would remove the code when the implementation will get merged. I will add a comment to it

File: CriticalMass/APIRequest.swift (hunk 1 above)
Comment (mltbnz) [reply CriticalMass/APIRequest.swift]: no reason. could be a property as well

File: CriticalMass/JSONDecoder+Additions.swift [hunk 3]
```diff
@@ -0,0 +1,11 @@
+//
+//  CriticalMaps
+import Foundation
+
+extension JSONDecoder {
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/JSONDecoder+Additions.swift]: fine with me. But maybe we want to move all extensions in an `Extensions` group? I find it harder to have an overview of what it is already there

[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/JSONDecoder+Additions.swift]: Would be a seperate PR

File: CriticalMass/APIRequest.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/APIRequest.swift]: Oh. I was not aware of that

Comment (mltbnz) [reply CriticalMass/APIRequest.swift]: I updated it to a property

**Pull request merged**